middleware/cors/cors.go

```diff
@@ -227,6 +233,12 @@ func New(config ...Config) fiber.Handler {
 			c.Set(fiber.HeaderAccessControlMaxAge, "0")
 		}
 
+		// Set Preflight request handling
+		if cfg.AllowPrivateNetworkAccess {
```

Comment:
I think this condition is wrong, looking at https://github.com/rs/cors/blob/master/cors.go#L380 they check if the `Request` has the Header set before setting it in the `response`.